python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -0,0 +1,419 @@
+    else:
+        with pytest.raises(expected_exception):
+            validate_domain(namespace)
+
+
+test_validate_pull_secret_data = [
+    ("should not raise any exception when namespace.pull_secret is None", Mock(pull_secret=None), None),
+    ("should not raise any exception when namespace.pull_secret is a valid JSON", Mock(pull_secret='{"key":"value"}'), None),
+    ("should raise InvalidArgumentValueError exception when namespace.pull_secret is not a valid JSON beacuse is an empty string", Mock(pull_secret=""), InvalidArgumentValueError),
+    ("should raise InvalidArgumentValueError exception when namespace.pull_secret is not a valid JSON because missing value", Mock(pull_secret='{"key": }'), InvalidArgumentValueError),
+    ("should raise InvalidArgumentValueError exception when namespace.pull_secret is not a valid JSON because is a simple string", Mock(pull_secret='a simple string'), InvalidArgumentValueError)
+]
+
+
+@pytest.mark.parametrize("test_description, namespace, expected_exception", test_validate_pull_secret_data, ids=[i[0] for i in test_validate_pull_secret_data])
+def test_validate_pull_secret(test_description, namespace, expected_exception):
+    if expected_exception is None:
+        validate_pull_secret(namespace)
+    else:
+        with pytest.raises(expected_exception):
+            validate_pull_secret(namespace)
+
+
+test_validate_sdn_data = [
+    ("should not raise any exception when namespace.software_defined_network is None", Mock(software_defined_network=None), None),
+    ("should not raise any exception when namespace.software_defined_network is OVNKubernetes", Mock(software_defined_network="OVNKubernetes"), None),
+    ("should not raise any exception when namespace.software_defined_network is OpenshiftSDN", Mock(software_defined_network="OpenshiftSDN"), None),
+    ("should raise InvalidArgumentValueError exception when namespace.software_defined_network is 'uknown', not one of the target values", Mock(software_defined_network="uknown"), InvalidArgumentValueError),
+    ("should raise InvalidArgumentValueError exception when namespace.software_defined_network is an empty string", Mock(software_defined_network=""), InvalidArgumentValueError)
+]
+
+
+@pytest.mark.parametrize("test_description, namespace, expected_exception", test_validate_sdn_data, ids=[i[0] for i in test_validate_sdn_data])
+def test_validate_sdn(test_description, namespace, expected_exception):
+    if expected_exception is None:
+        validate_sdn(namespace)
+    else:
+        with pytest.raises(expected_exception):
+            validate_sdn(namespace)
+
+
+test_validate_subnet_data = [
+    (
+        "should raise RequiredArgumentMissingError exception when subnet is not a valid resource id and namespace.vnet is False",
+        Mock(key='192.168.0.0/28', vnet=False),
+        'key',
+        False,
+        None,
+        None,
+        None,
+        None,
+        RequiredArgumentMissingError
+    ),
```

Comment:
aha! So we can format like this - can we spread it to every test case above and below? I find this much easier to read - I know where each test begins and ends.

Reply:
Done it.